### ISSUE TITLE
Refactor like_by, follow_by, comment_by => interact_by

### ISSUE THREAD

**Issue by timgrossmann** [InstaPy/InstaPy#4013]
#### In order to make InstaPy easier to use we want to cut down on the amount of ways things can be done. One of the ways to start making it simpler is to remove the separate methods `like_by_...`, `comment_by_...` and  `follow_by_...` and introducing a common "interface" as `interact_by_...`.

Advantages of this feature:
- Clean up code base
- Less places to fix problems if elements change
- Deduplication of code 
- Simpler documentation
- Higher level of abstraction
- Unified interface for users (always `interact_by` followed by e.g. `tags`, `users` etc.) and then controlling interactions by using `set_do_like`, `set_do_comment` and `set_do_follow`

Possible ways of achieving:
- Use the already existing methods as a "base line" on how this can be achieved 

> [Back this Issue](https://opencollective.com/instapy)

--- 

###### Feel free to comment with adjustments and improvements

**Comment by tE3m** [comment]
So would `set_do_follow` or other settings be global or only apply to a certain function? I think it'd be interesting to make settings callable from inside of one of those `interact_by` functions. What do you think?

**Comment by timgrossmann** [comment]
@TarekJ03 Sorry I think I don't exactly understand what you're suggesting.

The idea, for now was to have:
- `interact_by_tags`
- `interact_by_feed`
- `interact_by_followers`
- etc.

with the "same" functionality we have at the moment, just combined into one method.
To avoid confusions like "what is the difference between follow_users_followers and interact_user_followers with do_follow".

**Comment by timgrossmann** [comment]
@TarekJ03 do you mean having the follow like and comment percentage for each method separately instead of global? 🤔

**Comment by tE3m** [comment]
@timgrossmann yes, exactly. Setting `set_do_follow`, `set_do_like` and `set_do_comment` for each method seperately.

**Comment by timgrossmann** [comment]
@TarekJ03 Interesting approach 🤔 
It might make sense to do that, I think we have to consider that after analysing how people use InstaPy once the GUI is available.
